assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -147,4 +150,48 @@ describe("DetourMap", () => {
     expect(screen.getByTitle("Detour Start")).not.toBeNull()
     expect(screen.queryByTitle("Detour End")).toBeNull()
   })
+
+  test("missed stops are filled in when detour is complete", async () => {
+    const stop = stopFactory.build()
+    jest.mocked(fetchDetourMissedStops).mockReturnValue(instantPromise([stop]))
+
+    const { container } = render(<DiversionPage />)
+
+    await fireEvent.click(
+      container.querySelector(".c-detour_map--original-route-shape")!
+    )
+
+    await fireEvent.click(
+      container.querySelector(".c-detour_map--original-route-shape")!
+    )
+
+    await waitFor(() =>
+      expect(screen.queryByText(stop.name)).toBeInTheDocument()
+    )
+
+    expect(screen.getByText(stop.name)).toBeInTheDocument()
```

Comment:
suggestion(non-blocking): AFAIK, `waitFor` will retry a `expect` until it succeeds, or it will fail the test, so the second assert is redundant

```suggestion
    await waitFor(() =>
      expect(screen.getByText(stop.name)).toBeInTheDocument()
    )
   
```